### PR TITLE
Modify S2259: Migrate to LaYC - null dereference

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S2259/javascript/rule.adoc [hunk 1]
```diff
@@ -26,7 +34,19 @@ let foo = null;
 console.log(foo?.bar);
 ----
 
+Another option is to check in a `if` condition the equality to `null` or `undefined`.
```
Comment (victor-diez-sonarsource) [inline rules/S2259/javascript/rule.adoc:37]: I would put this latest suggestion in before the optional chaining, putting the accent in the last (optional chaining) as it's the newest and cleanest

**Review by victor-diez-sonarsource — APPROVED** [review]

**Review by leonardo-pilastri-sonarsource — CHANGES_REQUESTED** [review]
Looks good to me, just a few formatting changes

File: rules/S2259/java/compliant-code.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
+Ensuring the variable `myObject` has a value resolve the issue:
```
Comment (leonardo-pilastri-sonarsource) [inline rules/S2259/java/compliant-code.adoc:1]:
```suggestion
Ensuring the variable `myObject` has a value resolves the issue:
```

File: rules/S2259/java/compliant-code.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+{
+  if (input == null)
+  {
+    return;
+  }
+  System.out.println(input.toString()); // Compliant: if 'input' is null, this is unreachable
+}
+----
+
+Ensuring that no unboxing of `null` value can happen resolve the issue
```
Comment (leonardo-pilastri-sonarsource) [inline rules/S2259/java/compliant-code.adoc:25]:
```suggestion
Ensuring that no unboxing of `null` value can happen resolves the issue
```

File: rules/S2259/java/compliant-code.adoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+[source,java,diff-id=3,diff-type=compliant]
+----
+public boolean method() {
+  Boolean boxed = true;
+  return boxed; // Compliant
+}
+----
+
+Ensuring that both `conn` and `stmt` are not `null` resolve the issue:
```
Comment (leonardo-pilastri-sonarsource) [inline rules/S2259/java/compliant-code.adoc:35]:
```suggestion
Ensuring that both `conn` and `stmt` are not `null` resolves the issue:
```

File: rules/S2259/java/compliant-code.adoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  if (stmt != null) {
+    stmt.close();  // Compliant
+  }
+  if (conn != null) {
+    conn.close();  // Compliant
+  }
+}
+----
+
+Checking the returned value of `getName()` resolve the issue:
```
Comment (leonardo-pilastri-sonarsource) [inline rules/S2259/java/compliant-code.adoc:57]:
```suggestion
Checking the returned value of `getName()` resolves the issue:
```

File: rules/S2259/java/compliant-code.adoc [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  String name = getName();
+  if (name != null) {
+    return name.length() == 0; // Compliant
+  } else {
+    // ...
+  }
+}
+----
+
+Ensuring that the provided `color` is not `null` resolve the issue:
```
Comment (leonardo-pilastri-sonarsource) [inline rules/S2259/java/compliant-code.adoc:74]:
```suggestion
Ensuring that the provided `color` is not `null` resolves the issue:
```

File: rules/S2259/java/noncompliant-code.adoc [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+{
+  if (input == null)
+  {
+    // ...
+  }
+  System.out.println(input.toString()); // Noncompliant
+}
+----
+
+The unboxing triggered in the return statement will throw an `NullPointerException`:
```
Comment (leonardo-pilastri-sonarsource) [inline rules/S2259/java/noncompliant-code.adoc:25]:
```suggestion
The unboxing triggered in the return statement will throw a `NullPointerException`:
```

**Review by leonardo-pilastri-sonarsource — APPROVED** [review]
Looks good, I think there are just some grammar typos but they are negligible

**Pull request merged**